### PR TITLE
r/emr_studio_session_mapping - new resource

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Output from acceptance testing:

```
$ make testacc TESTS=TestAccEMRStudioSessionMapping_ PKG=emr
--- PASS: TestAccEMRStudioSessionMapping_disappears (95.15s)
--- PASS: TestAccEMRStudioSessionMapping_basic (163.24s)
```

### CODE REVIEW

File: internal/service/emr/studio_session_mapping.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+	if v, ok := d.GetOk("identity_id"); ok {
+		input.IdentityId = aws.String(v.(string))
+		id = v.(string)
+	}
+
+	if v, ok := d.GetOk("identity_name"); ok {
+		input.IdentityName = aws.String(v.(string))
+		id = v.(string)
+	}
```
Comment (ewbankkit) [inline internal/service/emr/studio_session_mapping.go:80]: `id` is set in both of these blocks - Are `identity_id` and `identity_name` mutually exclusive (and one is required)?
Can capture via `ExactlyOneOf` on the attributes.

File: internal/service/emr/studio_session_mapping.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+		},
+	}
+}
+
+func resourceStudioSessionMappingCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).EMRConn
+
+	var id string
+	studioId := d.Get("studio_id").(string)
+	indetityType := d.Get("identity_type").(string)
```
Comment (ewbankkit) [inline internal/service/emr/studio_session_mapping.go:65]: Nit: `identityType`.

File: internal/service/emr/studio_session_mapping.go (hunk 1 above)
Comment (DrFaust92) [reply internal/service/emr/studio_session_mapping.go:80]: the get method returns both of them ill try again but i think it makes this validations fails

File: internal/service/emr/studio_session_mapping.go (hunk 2 above)
Comment (DrFaust92) [reply internal/service/emr/studio_session_mapping.go]: fixed

File: internal/service/emr/studio_session_mapping.go (hunk 1 above)
Comment (DrFaust92) [reply internal/service/emr/studio_session_mapping.go:82]: added, works. not sure what i did wrong before

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% AWS_IDENTITY_STORE_USER_ID=... make testacc TESTS=TestAccEMRStudioSessionMapping_ PKG=emr
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/emr/... -v -count 1 -parallel 20 -run='TestAccEMRStudioSessionMapping_' -timeout 180m
=== RUN   TestAccEMRStudioSessionMapping_basic
=== PAUSE TestAccEMRStudioSessionMapping_basic
=== RUN   TestAccEMRStudioSessionMapping_disappears
=== PAUSE TestAccEMRStudioSessionMapping_disappears
=== CONT  TestAccEMRStudioSessionMapping_basic
=== CONT  TestAccEMRStudioSessionMapping_disappears
--- PASS: TestAccEMRStudioSessionMapping_disappears (49.77s)
--- PASS: TestAccEMRStudioSessionMapping_basic (81.24s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/emr	87.007s
```

**Pull request merged**